pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go

```diff
@@ -202,7 +202,13 @@ func (c *cyclonedxParser) GetPredicates(ctx context.Context) *assembler.IngestPr
 
 	preds := &assembler.IngestPredicates{}
 
-	toplevel := c.getPackageElement(string(c.cdxBom.Metadata.Component.BOMRef))
+	var toplevel []*model.PkgInputSpec = nil
+
+	if c.cdxBom.Metadata.Component != nil && c.cdxBom.Metadata.Component.BOMRef != "" {
```

Comment:
Hello @kurt-r2c! Thanks for the PR! This is something that warrants further discussion on how it should be handled in GUAC.

Reply:
I am wondering what the use case of having a CDX document without the top level component is for... if it is the case where it is interpretted as here's a list of all the components I use, then it should parse to a bunch of singletons, however, if it is not , then my thought would be to either use the generic `guac/cdx/<name>` or to just reject it. let's make sure we have this on the priority for triage.